src/sysext/sysext.c

```diff
@@ -516,6 +600,95 @@ static int mount_overlayfs(
         return 0;
 }
 
+static char *hierarchy_as_single_path_component(const char *hierarchy) {
+        /* We normally expect hierarchy to be /usr, /opt or /etc, but for debugging purposes the hierarchy
+         * could very well be like /foo/bar/baz/. So for a given hierarchy we generate a directory name by
+         * stripping the leading and trailing separators and replacing the rest of separators with dots. This
+         * makes the generated name to be the same for /foo/bar/baz and for /foo/bar.baz, but, again,
+         * speciyfing a different hierarchy is a debugging feature, so non-unique mapping should not be an
+         * issue in general case. */
```

Comment:
you could just use xescape() which should always work

Reply:
I can't see how xescape could help me here. I'm basically trying to convert `/usr` to `usr` or `/yadda/yadda/debug/hierarchy` to `yadda.yadda.debug.hierarchy` in order to generate the qualified path like `/var/lib/extension-data/usr.local` or `/var/lib/extension-data/yadda.yadda.debug.hierarchy.local` as specified in [the image extension update proposal](https://github.com/uapi-group/specifications/pull/78).